test/Network/IpAddress.cpp

```diff
@@ -0,0 +1,223 @@
+
+        SUBCASE("Byte constructor")
+        {
+            const sf::IpAddress ipAddress(142, 250, 69, 238);
+            CHECK(ipAddress.toString() == "142.250.69.238"s);
+            CHECK(ipAddress.toInteger() == 0x8EFA45EE);
+            CHECK(ipAddress != sf::IpAddress::None);
+        }
+
+        SUBCASE("Uint32 constructor")
+        {
+            const sf::IpAddress ipAddress(0xDEADBEEF);
+            CHECK(ipAddress.toString() == "222.173.190.239"s);
+            CHECK(ipAddress.toInteger() == 0xDEADBEEF);
+            CHECK(ipAddress != sf::IpAddress::None);
+        }
+    }
+
+    SUBCASE("Static functions")
+    {
+        SUBCASE("getLocalAddress")
+        {
+            const sf::IpAddress ipAddress = sf::IpAddress::getLocalAddress();
+            CHECK(ipAddress.toString() != "0.0.0.0");
+            CHECK(ipAddress.toInteger() != 0);
+            CHECK(ipAddress != sf::IpAddress::None);
+        }
+
+        SUBCASE("getPublicAddress")
+        {
+            const sf::IpAddress ipAddress = sf::IpAddress::getPublicAddress();
+            CHECK(ipAddress.toString() != "0.0.0.0");
+            CHECK(ipAddress.toInteger() != 0);
+            CHECK(ipAddress != sf::IpAddress::None);
+        }
+    }
+
+    SUBCASE("Static constants")
+    {
+        CHECK(sf::IpAddress::None.toString() == "0.0.0.0"s);
+        CHECK(sf::IpAddress::None.toInteger() == 0);
+
+        CHECK(sf::IpAddress::Any.toString() == "0.0.0.0"s);
+        CHECK(sf::IpAddress::Any.toInteger() == 0);
+
+        CHECK(sf::IpAddress::LocalHost.toString() == "127.0.0.1"s);
+        CHECK(sf::IpAddress::LocalHost.toInteger() == 0x7F000001);
+
+        CHECK(sf::IpAddress::Broadcast.toString() == "255.255.255.255"s);
+        CHECK(sf::IpAddress::Broadcast.toInteger() == 0xFFFFFFFF);
+    }
+
+    SUBCASE("Operators")
+    {
+        SUBCASE("operator==")
+        {
+            CHECK(sf::IpAddress() == sf::IpAddress());
+            CHECK(sf::IpAddress("") == sf::IpAddress(""s));
+            CHECK(sf::IpAddress("8.8.8.8") == sf::IpAddress(8, 8, 8, 8));
+            CHECK(sf::IpAddress(0x42, 0x69, 0x96, 0x24) == sf::IpAddress(0x42699624));
+            CHECK(sf::IpAddress(0xABCDEF01) == sf::IpAddress("171.205.239.1"));
+        }
+
+        SUBCASE("operator!=")
+        {
+            CHECK(sf::IpAddress() != sf::IpAddress("1.1.1.1"));
+            CHECK(sf::IpAddress(0x12344321) != sf::IpAddress(""));
+            CHECK(sf::IpAddress(192, 168, 1, 10) != sf::IpAddress(192, 168, 1, 11));
+        }
+
+        SUBCASE("operator<")
+        {
+            CHECK(sf::IpAddress(1) < sf::IpAddress(2));
+            CHECK(sf::IpAddress() < sf::IpAddress(0, 0, 0, 0));
+            CHECK(sf::IpAddress(0, 0, 0, 0) < sf::IpAddress(1, 0, 0, 0));
+            CHECK(sf::IpAddress(1, 0, 0, 0) < sf::IpAddress(0, 1, 0, 0));
+            CHECK(sf::IpAddress(0, 1, 0, 0) < sf::IpAddress(0, 0, 1, 0));
+            CHECK(sf::IpAddress(0, 0, 1, 0) < sf::IpAddress(0, 0, 0, 1));
```

Comment:
Isn't it surprising? I would have expected the opposite, the lexicographical order.
This current behavior is caused by the fact that the address is stored in network byte order (big-endian) in `m_address` and used like that in `operator<`.
A quick fix to get lexicographical ordering, if we agree this is wanted, would be to use the address converted in host byte order calling `ntohl` in `operator<`, but I think it would make more sense to store the address in host byte order instead.

Reply:
It definitely surprised me. I started writing the tests the other way around and was worried I found a bug when it didn't behave like I expected.

Reply:
I'm with @kimci86 here and don't see a reason to sort anti-lexicographically 🙂 
Also, SFML 3 would be the place to fix this, as it's quite a bit of a breaking change, even if rarely encountered.

Reply:
Want to fix that in this same PR or save it for later?

Reply:
What is `operator<` used for in this context besides sorting in `std::map`? If that's the main use case, the implementation of `operator<` doesn't really matter and most Network module users will never notice.

Reply:
It can be used for anything, like `std::sort()`, `std::binary_search()`, etc.

It probably does not matter in many cases, but if it's easy to fix, we might as well make it more intuitive for the few people where it does matter. I would say fixes to `sf::IpAddress` belong to a separate PR.